### PR TITLE
deprecate shard cache immediately if query failed

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
issue #22554

### CODE REVIEW

**Review by jiaoew1991 — APPROVED** [review]
/lgtm

File: internal/proxy/task_statistic.go [hunk 1]
```diff
@@ -304,24 +304,28 @@ func (g *getStatisticsTask) getStatisticsShard(ctx context.Context, nodeID int64
 		DmlChannels: channelIDs,
 		Scope:       querypb.DataScope_All,
 	}
+	log.Info("xxxx")
```
Comment (czs007) [inline internal/proxy/task_statistic.go:307]: what's this ?

**Pull request merged**